table/column.go

```diff
@@ -191,6 +191,34 @@ func CastValue(ctx sessionctx.Context, val types.Datum, col *model.ColumnInfo, r
 		truncateTrailingSpaces(&casted)
 	}
 
+	if (sc.InInsertStmt || sc.InUpdateStmt) && !casted.IsNull() && (col.Tp == mysql.TypeDate || col.Tp == mysql.TypeDatetime) {
```

Comment:
I think we should also consider `mysql.TypeTimestamp`.

Reply:
Thanks, added.